pyrk/th_component.py

```diff
@@ -35,27 +42,68 @@ def __init__(self, name=None,
         :type timer: Timer object
         :param heatgen: is this component a heat generator (fuel)
         :type heatgen: bool
+        :param power_tot: power generated in this component
+        :type power_tot: float
+        :param sph: is this component a spherical component, spherical equations
+        for heatgen, conduction are different, post-processing is different too
+        :type sph: bool
+        :param ri: inner radius of the sph/annular component, ri=0 for sphere
+        :type ri: float
+        :param ro: outer radius of the sph/annular component, ro=radius for sphere
+        :type ro: float
         """
         self.name = name
-        self.vol = vol.to('meter**3')
-        validation.validate_ge("vol", vol, 0*units.meter**3)
+        self.vol = vol
+        self.mat = mat
         self.k = mat.k
         self.cp = mat.cp
         self.dm = mat.dm
-        self.T0 = T0.to('kelvin')
-        validation.validate_num("T", T0)
+        self.timer = timer
         self.T = units.Quantity(np.zeros(shape=(timer.timesteps(),),
                                          dtype=float), 'kelvin')
         self.T[0] = T0
+        self.T0 = T0
         self.alpha_temp = alpha_temp.to('delta_k/kelvin')
-        self.timer = timer
         self.heatgen = heatgen
         self.power_tot = power_tot
         self.cond = {}
         self.conv = {}
+        self.adv = {}
         self.mass = {}
         self.cust = {}
         self.prev_t_idx = 0
+        self.convBC = {}
+        self.sph = sph
+        self.ri = ri
+        self.ro = ro
+
+    def mesh(self, size):
+        '''cut a THComponent into a list of smaller components
+        uniform meshing method, only implemented for spherical components
+        :param size: size of uniform mesh element
+        :type size: float.
+        :return: list of components
+        '''
+        assert self.sph, '''mesh function only implemented for spherical component'''
```

Comment:
ah! very good! you're quite right. we should take your input file out of
the main folder.

On Fri, Sep 4, 2015 at 4:35 PM, Xin Wang <notifications@github.com> wrote:

> In pyrk/th_component.py
> <https://github.com/pyrk/pyrk/pull/10#discussion_r38802830>:
>
> >          self.mass = {}
> >          self.cust = {}
> >          self.prev_t_idx = 0
> > +        self.convBC = {}
> > +        self.sph = sph
> > +        self.ri = ri
> > +        self.ro = ro
> > +
> > +    def mesh(self, size):
> > +        '''cut a THComponent into a list of smaller components
> > +        uniform meshing method, only implemented for spherical components
> > +        :param size: size of uniform mesh element
> > +        :type size: float.
> > +        :return: list of components
> > +        '''
> > +        assert self.sph, '''mesh function only implemented for spherical component'''
>
> It must be reading the wrong input file. Because I put my input file in
> the main folder.
> … <#14f9ab772d1f5e4d_>
> On Fri, Sep 4, 2015 at 4:30 PM, Katy Huff <notifications@github.com>
> wrote: In pyrk/th_component.py <
> https://github.com/pyrk/pyrk/pull/10#discussion_r38802601>: > self.mass =
> {} > self.cust = {} > self.prev_t_idx = 0 > + self.convBC = {} > + self.sph
> = sph > + self.ri = ri > + self.ro = ro > + > + def mesh(self, size): > +
> '''cut a THComponent into a list of smaller components > + uniform meshing
> method, only implemented for spherical components > + :param size: size of
> uniform mesh element > + :type size: float. > + :return: list of components
> > + ''' > + assert self.sph, '''mesh function only implemented for
> spherical component''' Unfortunately, not leftover files. khuff@hypatia:~/repos/pyrk/examples/sfr(withUnits)
> $ rm -rf out/* khuff@hypatia:~/repos/pyrk/examples/sfr(withUnits) $
> python ../../pyrk/driver.py --infile=input --plotdir=out 2015-09-04
> 15:46:07,392 - pyrklog - CRITICAL Welcome to PyRK. (c) Kathryn D. Huff Your
> simulation is starting. Perhaps it's time for a coffee.
> ,,,,yggyQQQQQQQQQQQQQQQQQyyyy,,,,
> ,,gQQQQQQQQQQQQ#WWWEEEEERRRREEEEWWW#QQQQQQQQQQQQQy,,
> ,gQQQQQQWR"D^^.-~~'~~WW~-..aQ ,},,....... ,.``?RW@QQQQQQ, ,QQQWR^.
> ,`Bm-gj^@L <https://github.com/L> .gyy, ``Q~-,,,a~^^ ^"WQQQQ @QQQy ,-~^`
> jQ,,DMmnmGQ`|Q `"~. ,#QQQ, @QQQQQQQQy,, ````` Q #`
> ````^""^^,,y#QQQQQQQQQQQQQg @QQQQQQQQQQQQQQQQQQQyyyy,,QQ@Q,,,,,,#Q,,,,ygg#QQQQQQQQQQQQQQQQQQQQQQQQQQQQ
> ]QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ"RRQQQQQQQQ
> QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ `@QQQQQQ
> <https://github.com/QQQQQQ>
> QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQM "QQQQQQ
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ ]QQQQQ
> 'QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQM ]QQQQQ
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ QQQQQ
> QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQM ,QQQQQ^
> ]QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ ,gQQQQQR
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQyQQQQQQQQQM^
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ#WR^`
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ^
> @QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ~
> "QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQE
> YQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ#^
> 'YQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ#T
> ?@QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQR `?RQQQQQQQQQQQQQQQQQQQQQQQQQQQ#R^
> 7RRQQQQQQQQQQQQQQQQQWE"`. 2015-09-04 16:27:07,017 - pyrklog - CRITICAL
> Simulation succeeded. khuff@hypatia:~/repos/pyrk/examples/sfr(withUnits)
> $ ls -l out/ total 43600 -rw-rw-r-- 1 khuff staff 244646 Sep 4 16:27 cool
> Temp[K].eps -rw-rw-r-- 1 khuff staff 78300 Sep 4 16:27 cool Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 244827 Sep 4 16:26 fuel0 Temp[K].eps -rw-rw-r-- 1
> khuff staff 77577 Sep 4 16:26 fuel0 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 244998 Sep 4 16:26 fuel1 Temp[K].eps -rw-rw-r-- 1 khuff staff 77504 Sep 4
> 16:26 fuel1 Temp[K].pdf -rw-rw-r-- 1 khuff staff 244828 Sep 4 16:26 fuel2
> Temp[K].eps -rw-rw-r-- 1 khuff staff 77475 Sep 4 16:26 fuel2 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245333 Sep 4 16:25 mod0 Temp[K].eps -rw-rw-r-- 1
> khuff staff 78870 Sep 4 16:25 mod0 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245327 Sep 4 16:25 mod1 Temp[K].eps -rw-rw-r-- 1 khuff staff 79056 Sep 4
> 16:25 mod1 Temp[K].pdf -rw-rw-r-- 1 khuff staff 245374 Sep 4 16:26 mod10
> Temp[K].eps -rw-rw-r-- 1 khuff staff 78968 Sep 4 16:26 mod10 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245351 Sep 4 16:26 mod11 Temp[K].eps -rw-rw-r-- 1
> khuff staff 78981 Sep 4 16:26 mod11 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245342 Sep 4 16:26 mod12 Temp[K].eps -rw-rw-r-- 1 khuff staff 78838 Sep 4
> 16:26 mod12 Temp[K].pdf -rw-rw-r-- 1 khuff staff 245384 Sep 4 16:26 mod13
> Temp[K].eps -rw-rw-r-- 1 khuff staff 78886 Sep 4 16:26 mod13 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245604 Sep 4 16:26 mod14 Temp[K].eps -rw-rw-r-- 1
> khuff staff 79167 Sep 4 16:26 mod14 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245398 Sep 4 16:26 mod15 Temp[K].eps -rw-rw-r-- 1 khuff staff 78832 Sep 4
> 16:26 mod15 Temp[K].pdf -rw-rw-r-- 1 khuff staff 246063 Sep 4 16:26 mod16
> Temp[K].eps -rw-rw-r-- 1 khuff staff 79135 Sep 4 16:26 mod16 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245376 Sep 4 16:26 mod17 Temp[K].eps -rw-rw-r-- 1
> khuff staff 78976 Sep 4 16:26 mod17 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 246215 Sep 4 16:26 mod18 Temp[K].eps -rw-rw-r-- 1 khuff staff 79200 Sep 4
> 16:26 mod18 Temp[K].pdf -rw-rw-r-- 1 khuff staff 246083 Sep 4 16:26 mod19
> Temp[K].eps -rw-rw-r-- 1 khuff staff 79096 Sep 4 16:26 mod19 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245344 Sep 4 16:25 mod2 Temp[K].eps -rw-rw-r-- 1
> khuff staff 78923 Sep 4 16:25 mod2 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245345 Sep 4 16:26 mod20 Temp[K].eps -rw-rw-r-- 1 khuff staff 78679 Sep 4
> 16:26 mod20 Temp[K].pdf -rw-rw-r-- 1 khuff staff 245359 Sep 4 16:26 mod21
> Temp[K].eps -rw-rw-r-- 1 khuff staff 78464 Sep 4 16:26 mod21 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245390 Sep 4 16:26 mod22 Temp[K].eps -rw-rw-r-- 1
> khuff staff 78462 Sep 4 16:26 mod22 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245202 Sep 4 16:26 mod23 Temp[K].eps -rw-rw-r-- 1 khuff staff 78280 Sep 4
> 16:26 mod23 Temp[K].pdf -rw-rw-r-- 1 khuff staff 245575 Sep 4 16:26 mod24
> Temp[K].eps -rw-rw-r-- 1 khuff staff 78195 Sep 4 16:26 mod24 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245342 Sep 4 16:25 mod3 Temp[K].eps -rw-rw-r-- 1
> khuff staff 79037 Sep 4 16:25 mod3 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245495 Sep 4 16:25 mod4 Temp[K].eps -rw-rw-r-- 1 khuff staff 79253 Sep 4
> 16:25 mod4 Temp[K].pdf -rw-rw-r-- 1 khuff staff 245226 Sep 4 16:25 mod5
> Temp[K].eps -rw-rw-r-- 1 khuff staff 79044 Sep 4 16:25 mod5 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 246081 Sep 4 16:26 mod6 Temp[K].eps -rw-rw-r-- 1
> khuff staff 79473 Sep 4 16:26 mod6 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 245426 Sep 4 16:26 mod7 Temp[K].eps -rw-rw-r-- 1 khuff staff 79199 Sep 4
> 16:26 mod7 Temp[K].pdf -rw-rw-r-- 1 khuff staff 246190 Sep 4 16:26 mod8
> Temp[K].eps -rw-rw-r-- 1 khuff staff 79414 Sep 4 16:26 mod8 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 245932 Sep 4 16:26 mod9 Temp[K].eps -rw-rw-r-- 1
> khuff staff 79299 Sep 4 16:26 mod9 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 247118 Sep 4 16:27 pebble Temp[K].eps -rw-rw-r-- 1 khuff staff 12667 Sep 4
> 16:27 pebble Temp[K].pdf -rw-rw-r-- 1 khuff staff 478224 Sep 4 16:24
> pow_and_rho.eps -rw-rw-r-- 1 khuff staff 66692 Sep 4 16:24 pow_and_rho.pdf
> -rw-rw-r-- 1 khuff staff 234646 Sep 4 16:24 power.eps -rw-rw-r-- 1 khuff
> staff 39544 Sep 4 16:24 power.pdf -rw-rw-r-- 1 khuff staff 240228 Sep 4
> 16:24 reactivity.eps -rw-rw-r-- 1 khuff staff 44909 Sep 4 16:24
> reactivity.pdf -rw-rw-r-- 1 khuff staff 245320 Sep 4 16:26 shell0
> Temp[K].eps -rw-rw-r-- 1 khuff staff 77735 Sep 4 16:26 shell0 Temp[K].pdf
> -rw-rw-r-- 1 khuff staff 246926 Sep 4 16:27 shell1 Temp[K].eps -rw-rw-r-- 1
> khuff staff 12586 Sep 4 16:27 shell1 Temp[K].pdf -rw-rw-r-- 1 khuff staff
> 7236011 Sep 4 16:25 temps.eps -rw-rw-r-- 1 khuff staff 2022395 Sep 4 16:25
> temps.pdf -rw-rw-r-- 1 khuff staff 1390460 Sep 4 16:24 zetas.eps -rw-rw-r--
> 1 khuff staff 181134 Sep 4 16:24 zetas.pdf khuff@hypatia:~/repos/pyrk/examples/sfr(withUnits)
> $ — Reply to this email directly or view it on GitHub <
> https://github.com/pyrk/pyrk/pull/10/files#r38802601>.
> -- ----------------------------------------- Xin WANG | 王昕 PhD Student
> Thermal Hydraulics Laboratory Nuclear Engineering Department University of
> California, Berkeley (415)408-1691
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyrk/pyrk/pull/10/files#r38802830>.
>



-- 
http://katyhuff.github.com